cmd/kubectl-gadget/deploy.go

```diff
@@ -430,6 +465,45 @@ func runDeploy(cmd *cobra.Command, args []string) error {
 		isPullSecretPresent = true
 	}
 
+	var isPolicyControllerPresent bool
+	if verifyImage {
+		if _, err = k8sClient.CoreV1().Namespaces().Get(context.TODO(), "cosign-system", metav1.GetOptions{}); err == nil {
+			isPolicyControllerPresent = true
+		} else {
+			log.Warnf("No policy controller found, the container image will not be verified")
+		}
+
+		if isPolicyControllerPresent {
+			encodedKey := base64.StdEncoding.EncodeToString([]byte(publicKey))
+
+			// We cannot use tag as image for admission controller, as the tested image
+			// will use digest:
+			// Error: problem while creating resource: creating "DaemonSet": admission webhook "policy.sigstore.dev" denied the request: validation failed: no matching policies: spec.template.spec.containers[0].image
+			// ghcr.io/inspektor-gadget/inspektor-gadget@sha256:a6c2b00174013789d4af0cc48ba5e269426ff44f27dcb9b84f489537280e0871
+			// So, if users gives digest, we use it directly.
+			// Otherwise, i.e. user gives a tag or the image itself, we extract the
+			// repository from it and add "**" to glob it.
+			var admissionImage string
+			if strings.Contains(image, "@") {
+				admissionImage = image
+			} else {
+				parts := strings.Split(image, ":")
+				admissionImage = fmt.Sprintf("%s**", parts[0])
+			}
```

Comment:
I think this logic isn't complete enough. For instance an image like `host.com:5000/alpine` will erroneously  create a blob like `host.com**`. Perhaps you need to use something like [normalizeImageName](https://github.com/inspektor-gadget/inspektor-gadget/blob/e1e44334dc0f21083ac019859c105bcc3b474c49/pkg/oci/oci.go#L404-L405) and / or https://pkg.go.dev/github.com/distribution/reference to correctly parse the image.